5.5/docker-entrypoint.sh

```diff
@@ -1,54 +1,94 @@
 #!/bin/bash
 set -e
 
+get_option () {
+	local section=$1
+	local option=$2
+	local default=$3
+	ret=$(my_print_defaults $section | grep '^--'${option}'=' | cut -d= -f2-)
+	[ -z $ret ] && ret=$default
+	echo $ret
+}
+
 if [ "${1:0:1}" = '-' ]; then
 	set -- mysqld "$@"
 fi
 
 if [ "$1" = 'mysqld' ]; then
-	# read DATADIR from the MySQL config
+	# Get config
 	DATADIR="$("$@" --verbose --help 2>/dev/null | awk '$1 == "datadir" { print $2; exit }')"
-	
+	SOCKET=$(get_option  mysqld socket "/tmp/mysql.sock")
+	HOSTNAME=$(hostname)
+	PIDFILE=$(get_option mysqld pid-file "$DATADIR/$HOSTNAME.pid")
+
 	if [ ! -d "$DATADIR/mysql" ]; then
 		if [ -z "$MYSQL_ROOT_PASSWORD" -a -z "$MYSQL_ALLOW_EMPTY_PASSWORD" ]; then
 			echo >&2 'error: database is uninitialized and MYSQL_ROOT_PASSWORD not set'
 			echo >&2 '  Did you forget to add -e MYSQL_ROOT_PASSWORD=... ?'
 			exit 1
 		fi
-		
-		echo 'Running mysql_install_db ...'
-		mysql_install_db --datadir="$DATADIR" --basedir=/usr/local/mysql
+		if [ ! -d "$DATADIR" ]; then
+			mkdir -p $DATADIR
+		fi
+		chown -R mysql:mysql "$DATADIR"
+
+		echo 'Running mysql_install_db'
+		mysql_install_db --user=mysql --datadir=$DATADIR --rpm --basedir=/usr/local/mysql
 		echo 'Finished mysql_install_db'
-		
+
+		mysqld --user=mysql --datadir=$DATADIR --skip-networking --basedir=/usr/local/mysql &
+		for i in $(seq 30 -1 0); do
+			[ -S $SOCKET ] && break
+			echo 'MySQL init process in progress...'
+			sleep 1
+		done
+		if [ $i = 0 ]; then
+			echo >&2 'MySQL init process failed.'
+			exit 1
+		fi
+
 		# These statements _must_ be on individual lines, and _must_ end with
 		# semicolons (no line breaks or comments are permitted).
 		# TODO proper SQL escaping on ALL the things D:
-		
-		tempSqlFile='/tmp/mysql-first-time.sql'
+
+		tempSqlFile=$(mktemp /tmp/mysql-first-time.XXXXXX.sql)
 		cat > "$tempSqlFile" <<-EOSQL
 			DELETE FROM mysql.user ;
 			CREATE USER 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}' ;
 			GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
 			DROP DATABASE IF EXISTS test ;
 		EOSQL
-		
+
 		if [ "$MYSQL_DATABASE" ]; then
-			echo "CREATE DATABASE IF NOT EXISTS \`$MYSQL_DATABASE\` ;" >> "$tempSqlFile"
+			echo "CREATE DATABASE IF NOT EXISTS \`"$MYSQL_DATABASE"\` ;" >> "$tempSqlFile"
```

Comment:
Since this is already inside double quotes, why bother with ending them to drop in the variable? (alternatively, why not switch to single quotes instead so that the backtick doesn't have to be escaped?)